### PR TITLE
Generate ICHs benefitting from artifact anti-champion mods

### PR DESCRIPTION
**Pull request by robojumper**
This is for a DIM search filter that matches weapons with seasonal anti-champion mods in the artifact. Bungie keeps deprecating and re-adding the anti-champion mods every season with varying reliability, so I'm checking the artifact itself for safety.

### CODE REVIEW

**Review by bhollis — APPROVED** [review]

**Comment by bhollis** [comment]
Is this still valid with the current artifact configuration?

**Comment by nev-r** [comment]
needed defs dependency updates, re-running against current artifact, and a regex update.  
not sure this is that resilient, it relies on bungie writing consistent copy, and, hahahahahaha

**Comment by robojumper** [comment]
Yeah I was worried the regex would require tweaking each season anyway, but seems good to me.

**Comment by robojumper** [comment]
Okay, fixed it for this season (old mods weren't deprecated, new mods aren't gauntlet mods anymore)

**Comment by delphiactual** [comment]
This still works as of season 23

**Comment by delphiactual** [comment]
@robojumper updated for TFS

File: src/generate-artifact-breaker-weapons.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+const inventoryItems = getAllDefs('InventoryItem');
+
+// First, find the seasonal artifact
+const artifactVendor = inventoryItems.find(
+  (i) => i.itemTypeDisplayName?.includes('Artifact') && currentSeasonDef.hash === i.seasonHash,
+)!.preview!.previewVendorHash;
+const artifactMods = getDef('Vendor', artifactVendor)!
+  // Take the last 25 mods, since sometimes Bungie doesn't deprecate the old mods
+  // Had to increase this by 10 during TFS due to ??Classified?? mods?
```
Comment (delphiactual) [inline src/generate-artifact-breaker-weapons.ts:26]: we'll this was explained in the echoes reveal we get 2 more rows per act.

**Pull request merged**